app/src/main/java/com/myperishableplanner/v21001/ItemViewModel.kt

```diff
@@ -13,25 +13,28 @@ import com.google.firebase.firestore.FirebaseFirestore
 import com.myperishableplanner.v21001.dto.ItemDetail
 
 
-class ItemViewModel (var itemService: IItemService = ItemService()): ViewModel() {
+class ItemViewModel (var itemService: IItemService = ItemService()) : ViewModel() {
 
+    var items: MutableLiveData<List<Item>> = MutableLiveData<List<Item>>()
+    var searchText = "apple"
 
-        var items : MutableLiveData<List<Item>> = MutableLiveData<List<Item>>()
+    private lateinit var firestore: FirebaseFirestore
 
-        private lateinit var firestore : FirebaseFirestore
-
-        init {
-            {
-                firestore = FirebaseFirestore.getInstance()
-                firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
-            }
+    init {
+        {
+            firestore = FirebaseFirestore.getInstance()
+            firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
         }
+    }
 
-        fun fetchItems(){
-            viewModelScope.launch {
-                items.postValue(itemService.fetchItems())
+    fun fetchItems(){
+        viewModelScope.launch {
+            val results = itemService.fetchItems(searchText)
+            results?.let {
+                items.postValue(it)
```

Comment:
the fetchItems method on the ItemService class now takes a string parameter instead of a hard coded value.